packages/react-aria-components/docs/styling.mdx

```diff
@@ -0,0 +1,351 @@
+
+All states supported by each component are listed in the Styling section of their documentation.
+
+## Render props
+
+The `className` and `style` props also accept functions which receive states for styling. This lets you dynamically determine the classes or styles to apply.
+
+```jsx
+<Item className={({isSelected}) => isSelected ? 'selected' : 'unselected'}>
+  Item
+</Item>
+```
+
+Render props may also be used as children to alter what elements are rendered based on the current state. For example, you could render a checkmark icon when an item is selected.
+
+```jsx
+<Item>
+  {({isSelected}) => (
+    <>
+      {isSelected && <CheckmarkIcon />}
+      <span>Item</span>
+    </>
+  )}
+</Item>
+```
+
+The render props exposed for each component are listed in the Styling section of their documentation.
+
+## Slots
+
+Some components include multiple instances of the same component as children. These use the `slot` prop to distinguish them, which can also be used in CSS for styling purposes. This example targets the increment and decrement buttons in a [NumberField](NumberField.html).
+
+```tsx
+<NumberField>
+  <Label>Width</Label>
+  <Group>
+    <Input />
+    {/*- begin highlight -*/}
+    <Button slot="increment">+</Button>
+    <Button slot="decrement">-</Button>
+    {/*- end highlight -*/}
+  </Group>
+</NumberField>
+```
+
+```css render=false
+.react-aria-NumberField {
+  /*- begin highlight -*/
+  [slot=increment] { /*- end highlight -*/
+    border-radius: 4px 4px 0 0;
+  }
+
+  /*- begin highlight -*/
+  [slot=decrement] { /*- end highlight -*/
+    border-radius: 0 0 4px 4px;
+  }
+}
+```
+
+The slots supported by each component are shown in the Anatomy section of their documentation.
+
+## CSS variables
+
+Some components provide CSS variables that you can use in your styling code. For example, the [Select](Select.html) component provides a `--trigger-width` variable on the popover that is set to the width of the trigger button. You can use this to make the width of the popover match the width of the button.
+
+```css render=false
+.react-aria-Popover {
+  width: var(--trigger-width);
+}
+```
+
+The CSS variables provided by each component are listed in the Styling section of their documentation.
+
+## Tailwind CSS
+
+[Tailwind CSS](https://tailwindcss.com) is a utility-first CSS framework for rapid styling that works great with React Aria Components. To access [states](#states), you can use [data attributes](https://tailwindcss.com/docs/hover-focus-and-other-states#data-attributes) as modifiers:
+
+```jsx
+<Item className="data-[selected]:bg-blue-400 data-[disabled]:bg-gray-100">
+  Item
+</Item>
+```
+
+Alternatively, you can use [render props](#render-props) to control which Tailwind classes are applied based on states. This can be useful if you need to apply multiple clases based on a single state:
```

Comment:
```suggestion
Alternatively, you can use [render props](#render-props) to control which Tailwind classes are applied based on states. This can be useful if you need to apply multiple classes based on a single state:
```